src/Reporting/src/App.Metrics.Formatters.Prometheus/Internal/AsciiFormatter.cs

```diff
@@ -37,54 +37,54 @@ await using (var streamWriter = new StreamWriter(destination, Encoding, bufferSi
             {
                 foreach (var metricFamily in metricFamilies)
                 {
-                    WriteFamily(streamWriter, metricFamily);
+                    await WriteFamily(streamWriter, metricFamily);
                 }
 
                 await streamWriter.FlushAsync();
             }
         }
 
-        private static void WriteFamily(StreamWriter streamWriter, MetricFamily metricFamily)
+        private static async Task WriteFamily(StreamWriter streamWriter, MetricFamily metricFamily)
```

Comment:
Method here should be renamed `WriteFamilyAsync` and the signature should include `CancelationToken`